### PR TITLE
[CosmosDB] Changes for Rest API Version 2023-03-15-preview

### PR DESCRIPTION
**Pull request by SrinikhilReddy**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Type.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.CosmosDB.Models
+{
+    /// <summary> Type of the connection string. </summary>
+    public readonly partial struct Type : IEquatable<Type>
```
Comment (live1206) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Type.cs:14]: `Type` is too generic, consider renaming to `CosmosDBConnectionStringType`.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Status.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.CosmosDB.Models
+{
+    /// <summary> Enum to indicate current buildout status of the region. </summary>
+    public readonly partial struct Status : IEquatable<Status>
```
Comment (live1206) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Status.cs:14]: `Status` is too generic, consider renaming to `CosmosDBBuildoutStatus` or some other name makes sense.

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/CheckNameAvailabilityContent.Serialization.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.ResourceManager.CosmosDB.Models
+{
+    public partial class CheckNameAvailabilityContent : IUtf8JsonSerializable
```
Comment (live1206) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/CheckNameAvailabilityContent.Serialization.cs:13]: `CheckNameAvailabilityContent` is too generic, consider renaming to `CheckCosmosDBNameAvailabilityContent`
```yaml
CheckNameAvailabilityRequest: CheckCosmosDBNameAvailabilityContent
```

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/NodeGroupProperties.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            Sku = sku;
+            DiskSizeGB = diskSizeGB;
+            EnableHa = enableHa;
+        }
+
+        /// <summary> The resource sku for the node group. This defines the size of CPU and memory that is provisioned for each node. Example values: &apos;M30&apos;, &apos;M40&apos;. </summary>
+        public string Sku { get; set; }
+        /// <summary> The disk storage size for the node group in GB. Example values: 128, 256, 512, 1024. </summary>
+        public long? DiskSizeGB { get; set; }
```
Comment (live1206) [inline sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/NodeGroupProperties.cs:32]: consider renaming to `DiskSizeInGB`

**Review by live1206 — COMMENTED** [review]
Please follow the guide the fix CI failure:
```
Generated code is not up to date.
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code.
    run 'dotnet build /t:GenerateTests' to update the generated test code.
    
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory cosmosdb'.
```

Please update changelog with a new version and expecting release date, also update the version in `Azure.ResourceManager.CosmosDB.csproj`.

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/AzSdk.RP.props [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
   <PropertyGroup>
-    <AzureApiTag>CosmosDB_2022-08-15-preview</AzureApiTag>
+    <AzureApiTag>CosmosDB_2023-03-15-preview</AzureApiTag>
```
[the remark this answers is not in the document]
Comment (himanshudhawale) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/AzSdk.RP.props]: reverted

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/ApiCompatBaseline.txt [hunk 6]
```diff
@@ -0,0 +1 @@
+Total Issues: 0
```
[the remark this answers is not in the document]
Comment (himanshudhawale) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/ApiCompatBaseline.txt]: removed

File: sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/CosmosDBManagementClient.cs [hunk 7]
```diff
@@ -547,7 +547,7 @@ private void Initialize()
             RestorableTableResources = new RestorableTableResourcesOperations(this);
             Service = new ServiceOperations(this);
             BaseUri = new System.Uri("https://management.azure.com");
-            ApiVersion = "2022-08-15-preview";
+            ApiVersion = "2023-03-15-preview";
```
[the remark this answers is not in the document]
Comment (himanshudhawale) [reply sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/CosmosDBManagementClient.cs]: reverted

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Status.cs (hunk 2 above)
Comment (himanshudhawale) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Status.cs]: added CosmosDBStatus

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Type.cs (hunk 1 above)
Comment (himanshudhawale) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/Type.cs]: Renamed

File: sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/ProvisioningState.cs [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.CosmosDB.Models
+{
+    /// <summary> The provisioning state of the resource. </summary>
+    public readonly partial struct ProvisioningState : IEquatable<ProvisioningState>
```
[the remark this answers is not in the document]
Comment (himanshudhawale) [reply sdk/cosmosdb/Azure.ResourceManager.CosmosDB/src/Generated/Models/ProvisioningState.cs]: renamed

**Comment by SrinikhilReddy** [comment]
Hi @live1206 [live1206](https://github.com/live1206). We see that in the repo all test recordings were deleted in this PR. [Migrate to Test Proxy for some MGMT RPs. batch 3 by dvbb · Pull Request #37136 · Azure/azure-sdk-for-net (github.com)](https://github.com/Azure/azure-sdk-for-net/pull/37136/files)

Previously the expectation with new RP APIs was to re-record all tests, which this PR does. What is the expectation now? should we delete the latest recordings?

**Review by live1206 — COMMENTED** [review]
Please resolve the comments in [APIView](https://github.com/Azure/azure-sdk-for-net/pull/36950#issuecomment-1585076577) as well

**Review by live1206 — DISMISSED** [review]

**Review by live1206 — APPROVED** [review]

**Pull request merged**